src/document-processor.ts

```
@@ -0,0 +1,811 @@
+        loader = new PDFLoader(filePath);
+        break;
+      case 'docx':
+        loader = new DocxLoader(filePath);
+        break;
+      case 'txt':
+        loader = new TextLoader(filePath);
+        break;
+      default:
+        // For other formats, create a document from the content
+        const content = await this.getFileContent(filePath, fileType);
+        return [new Document({
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)